eng/helix/helix.proj

```diff
@@ -61,6 +76,35 @@
     <HelixProperties Condition="'$(RunQuarantinedTests)' != 'true'" Include="runType" Value="unquarantined" />
   </ItemGroup>
 
+  <Target Name="IncludeAspNetRuntime" BeforeTargets="Gather">
+    <MSBuild Projects="$(RepoRoot)src\Framework\App.Runtime\src\Microsoft.AspNetCore.App.Runtime.csproj"
+        Properties="DisableYarnCheck=true;ExcludeFromBuild=false"
+        Targets="_GetPackageVersionInfo"
+        SkipNonexistentProjects="false">
+      <Output TaskParameter="TargetOutputs" ItemName="_ResolvedProductVersionInfo" />
+    </MSBuild>
+
+    <PropertyGroup>
+      <SharedFxVersion>@(_ResolvedProductVersionInfo->'%(PackageVersion)')</SharedFxVersion>
+    </PropertyGroup>    
+    
+    <Unzip SourceFiles="$(RepoRoot)artifacts\packages\$(Configuration)\Shipping\Microsoft.AspNetCore.App.Runtime.$(TargetOsName)-$(TargetArchitecture).$(SharedFxVersion).nupkg" DestinationFolder="$(OutputPath)\AspNetCoreAppRuntimeHelixRaw" />
+    <ItemGroup>
+      <_appRuntimeFiles Include="$(OutputPath)\AspNetCoreAppRuntimeHelixRaw\**\*.txt" />
+      <_appRuntimeFiles Include="$(OutputPath)\AspNetCoreAppRuntimeHelixRaw\**\*.json" />
+      <_appRuntimeFiles Include="$(OutputPath)\AspNetCoreAppRuntimeHelixRaw\**\*.dll" />
+      <_appRuntimeFiles Include="$(OutputPath)\AspNetCoreAppRuntimeHelixRaw\**\*.xml" />
+    </ItemGroup>
+    <Copy SourceFiles="@(_appRuntimeFiles)" DestinationFolder="$(OutputPath)\AspNetCoreAppRuntimeHelix\shared\Microsoft.AspNetCore.App\$(SharedFxVersion)" />
+    <Unzip SourceFiles="$(RepoRoot)artifacts\packages\$(Configuration)\Shipping\Microsoft.AspNetCore.App.Ref.$(SharedFxVersion).nupkg" DestinationFolder="$(OutputPath)\AspNetCoreAppRuntimeHelix\packs\Microsoft.AspNetCore.App.Ref\$(SharedFxVersion)" />
```

Comment:
Similarly

```suggestion
    <ItemGroup>
     <_RefFilesToCopy Include="$(RepoRoot)/.dotnet/packs/Microsoft.AspNetCore.App.Ref/$(SharedFxVersion)/**/*" />
    </ItemGroup>

    <Copy SourceFiles="@(_RefFilesToCopy)" DestinationFiles="@(_RefFilesToCopy->'$(OutputPath)/AspNetCoreAppRuntimeHelix/packs/Microsoft.AspNetCore.App.Ref/$(SharedFxVersion)%(RecursiveDir)%(FileName)%(Extension)')" />
```

This is a bit more complicated because Microsoft.AspNetCore.App.Ref has multiple sub-folders. But, again, there's no need to `unzip` the .nupkg file.

Separately, this part should be conditional on `$(IsTargetingPackBuilding)` because that's not normally `true` after RTM. That property isn't available in this project because it's set in the root Directory.Build.props. So, you'll need to duplicate https://github.com/dotnet/aspnetcore/pull/31448/files#diff-6ba44602dd26317cbdb4fdd62bffbf78e7aba91fb130e82da0bbc973ed0f7f83R80-R89 but with Microsoft.AspNetCore.App.Ref.csproj and different item and property names. Then, do the copying only if ` '$(SharedFxVersion)' == '$(TargetingPackVersion)' `. Don't worry about build performance; the `_GetPackageVersionInfo` target is cheap.

Reply:
If we aren't worrying about build performance, everything you just described sounds way more complicated than just unzipping the nupkg since its exactly what we already need to include.  What's the benefit of copying around individual sub folders?

Reply:
The other runtime package is nice since we can just directly copy what we want into the payload, if the ref pack is not already built, I'd much rather keep the current logic which just unpacks the final package into the helix payload.  Its clean and simple

Reply:
Oh I didn't see the suggestion, ok I can just do the copy via your suggestion thanks

Reply:
What did you mean about IsTargetingPackBuilding?  What do we do if its not building exactly?  We don't need to include our own runtime correlation?

Reply:
I mean that, when `$(IsTargetingPackBuilding)` is _not_ `true`, there's probably no `.dotnet/packs/Microsoft.AspNetCore.App.Ref/$(SharedFxVersion)` folder at all. The previous code would have failed (loudly) because there was no `.nupkg` to `unzip`. This might be fine because `<Copy />` should simply no-op when there are no `@(_RefFilesToCopy)` items. But, it should be tested before this goes in.

The alternative is to add `Condition`s to confirm the targeting pack is building and comparing the package versions confirms that.

Reply:
What do we want to do if the targetpack isn't building?  Just no-op?  Seems fine to just add that condition to the ref packs if that's what you are concerned about

Reply:
Yes exactly

Reply:
IsTargetingPackBuilding isn't defined in helix.proj either, wouldn't these be no-ops anyways if there's nothing included in the ref files to copy?

Reply:
It _should_ no-op. That's why I suggested testing to make sure it'll work when that property is `false`. The other option of getting the package version from Microsoft.AspNetCore.App.Ref.csproj and using that in another `Condition` or two is a bit more complicated.

Reply:
I can just test it with a non existent directory to make sure it no-ops when no files are included